### ISSUE TITLE
Nested less dependencies do not live-reload

### ISSUE THREAD

**Issue by matthewp** [donejs/donejs#619]
### module.js

``` js
import "./module.less";
```
### module.less

``` less
@import "./other.less"
```

If **other.less** changes it will not be reflected in live-reload either on the client or server. Effectively meaning the only way to see the changes is to restart the server (or use development.html).

**Comment by matthewp** [comment]
The problem is that these nested less files are not part of the steal dependency graph; steal does not know about them.  The only way I can see to fix is to make all less files part of the dep graph, make them be treated as dependencies by adding an instantiate hook.

**Comment by akagomez** [comment]
@matthewp Will adding these files to Steal's dependency graph fix this issue also? https://github.com/stealjs/steal-less/issues/18

**Comment by matthewp** [comment]
I don't think so, I don't know why less is loading that file twice, could be a less bug or perhaps we are not implementing some sort of "this has already loaded" API, idk.

**Comment by justinbmeyer** [comment]
@matthewp I know there's an api to associate multiple modules with one file (for `.component`).  Perhaps there should be another API to associate multiple files with one module.

**Comment by matthewp** [comment]
We could use load metadata to list the files that should be associated. The watcher would then use this to know what the parent module is (the module that's actually part of the graph and needs to be re-executed).

**Comment by matthewp** [comment]
**Update**: changed live-reload and steal-tools so that the `includedDeps` metadata property is used to know of dependencies that are not part of the normal dependency graph. Plugins can add this metadata and we'll watch for these files and trigger reloads when they change, just like for normal deps.

Next need to update the less plugin so that it sets this metadata for its nested dependencies and test the entire flow.

**Comment by matthewp** [comment]
I have this working in a prerelease. If you update your dependencies to:

```js
"dependencies": {
  ...
  "steal": "pre"
},
"devDependencies": {
  ...
  "steal-tools": "pre"
}
```

You can try it out. Will wait on some feedback before releasing.

**Comment by matthewp** [comment]
I have this working in a prerelease. If you update your dependencies to:

``` js
"dependencies": {
  ...
  "steal": "pre"
},
"devDependencies": {
  ...
  "steal-tools": "pre"
}
```

You can try it out. Will wait on some feedback before releasing.

**Comment by matthewp** [comment]
cc @akagomez

**Comment by leoj3n** [comment]
@matthewp Seems to be working!

**Comment by akagomez** [comment]
Somehow I completely missed the chatter on this. I'm stoked to give this a try. You're my hero @matthewp.

**Comment by akagomez** [comment]
Doesn't seem to be working for me. Probably doing something wrong.

**Comment by matthewp** [comment]
Note, that this doesn't solve the multiple load problem, I tried to recreate that here: https://github.com/stealjs/steal-less/issues/18

**Comment by tracer99** [comment]
@matthewp this this ever get released?

**Comment by matthewp** [comment]
@tracer99 Yep.

**Comment by tracer99** [comment]
@matthewp which version of steel got it? We just updated to what we thought was the latest, but the issue still exists. We see the console show that it's detected the change of an imported less file, but the UI doesn't update until we restart the server.
